File: insertion-sort/insertion_sort.py
# File: intertion_sort.py
# Description: Practicing implementing basic data structures and algorithms
#              from scratch.
# Author: Victor Salcedo
# Date: December 30, 2025
# Contact: vsalc@proton.me

def insertionSort(arr):
  j = 1
  end = len(arr)

  while j < end:
    current = arr[j]
    i = j - 1
    while i >= 0 and arr[i] > current:
      arr[i + 1] = arr[i]
      i -= 1
    arr[i + 1] = current
    j += 1

File: insertion-sort/insertion-sort.js
/*
 * File: insertion-sort.js
 * Description: Implementing basic data structures and algorithms in 
 *              JavaScript.
 * Author: Victor Salcedo
 * Date: January 1, 2026
 * Contact: vsalc@proton.me
 */

function insertionSort(arr) {
  let j = 1
  const end = arr.length

  while (j < end) {
    const key = arr[j]
    // Insert A[j] into the sorted sequence A[0..j-1].
    let i = j - 1
    while (i >= 0 && arr[i] > key) {
      arr[i + 1] = arr[i]
      i -= 1
    }
    arr[i + 1] = key
    j += 1
  }
}

export default insertionSort